### ISSUE TITLE
ENH: Refactoring or look and feel improvements of STL procedure (_stl.pyx)

### ISSUE THREAD

**Issue by 1pyxa1** [statsmodels/statsmodels#8898]
I would like to address the issues I encountered while trying to understand and follow the code of the STL procedure. The lack of comments and the presence of numerous inexpressive variables made it difficult for me to grasp the logic and flow of the code.

I am unsure whether it would be beneficial to partially rewrite the computational aspects of the procedure or focus only on improving its overall look and feel. In this issue, I propose two code refactoring variants: one involving a partial rewrite of the  implementation and the other focusing on enhancing readability and usability.

Variant 1: Partial Rewriting
- In this variant is review and rewrite some patterns of the STL procedure.
- For example, make some methods static with passing attributes, remove this "work" variable that is really hard to follow (in my opinion), fix all this `work1 = self._work[2, :]` and so on. 

Variant 2: Readability and Usability Enhancements
- In this variant, the primary focus would be on improving the readability and usability of the STL procedure.
- This approach would involve adding comments, providing clear explanations of the code logic, and utilizing expressive variable names.
		
I would like to contribute with PR, but I want to make sure that it is in demand and understand how best to do it.

**Comment by bashtage** [comment]
It is a literal translation of the original source.  This was done to ensure correctness and fidelity to the Fortran source.  Is there a long run goal, aside from rewriting?  

If you want to understand STL from source, you should take a look at the Fortran.

**Comment by 1pyxa1** [comment]
thanks for the reply! no, there is no any long run goals. just wanted to add a bit more readability (as i can see it).
